modules/atmosphere/rendering/renderableatmosphere.h

```diff
@@ -0,0 +1,154 @@
+    std::pair<std::string, double> caster;
+};
+
+struct ShadowRenderingStruct {
+    double xu,
+           xp;
+    double rs,
+           rc;
+    glm::dvec3 sourceCasterVec;
+    glm::dvec3 casterPositionVec;
+    bool isShadowing;
+};
+
+namespace planetgeometry {
+class PlanetGeometry;
+}
+
+namespace documentation { struct Documentation; }
+namespace planetgeometry { class PlanetGeometry; }
+
+class RenderableAtmosphere : public Renderable {
+public:    
+
+    RenderableAtmosphere(const ghoul::Dictionary& dictionary);
+
+    void deinitialize() override;
+    void initializeGL() override;
+    void deinitializeGL() override;
+    bool isReady() const override;
+
+    void render(const RenderData& data, RendererTasks& rendererTask) override;
+    void update(const UpdateData& data) override;
+
+    static documentation::Documentation Documentation();
+
+private: 
+    glm::dmat4 computeModelTransformMatrix(const openspace::TransformData& transformData);
+    void updateAtmosphereParameters();
+        
+    properties::FloatProperty _atmosphereHeightP;
+    properties::FloatProperty _groundAverageReflectanceP;
+    properties::FloatProperty _groundRadianceEmittionP;
+    properties::FloatProperty _rayleighHeightScaleP;
+    properties::FloatProperty _rayleighScatteringCoeffXP;
+    properties::FloatProperty _rayleighScatteringCoeffYP;
+    properties::FloatProperty _rayleighScatteringCoeffZP;
+    properties::BoolProperty  _ozoneEnabledP;
+    properties::FloatProperty _ozoneHeightScaleP;
+    properties::FloatProperty _ozoneCoeffXP;
+    properties::FloatProperty _ozoneCoeffYP;
+    properties::FloatProperty _ozoneCoeffZP;
+    properties::FloatProperty _mieHeightScaleP;
+    properties::FloatProperty _mieScatteringCoeffXP;
+    properties::FloatProperty _mieScatteringCoeffYP;
+    properties::FloatProperty _mieScatteringCoeffZP;
+    properties::FloatProperty _mieScatteringExtinctionPropCoefficientP;
+    properties::FloatProperty _mieAsymmetricFactorGP;
+    properties::FloatProperty _sunIntensityP;
+    properties::BoolProperty  _sunFollowingCameraEnabledP;
+    properties::BoolProperty  _hardShadowsEnabledP;
+
+    bool _atmosphereEnabled;
+    bool _ozoneLayerEnabled;
+    bool _sunFollowingCameraEnabled;   
+    float _atmosphereRadius;
+    float _atmospherePlanetRadius;
+    float _planetAverageGroundReflectance;
+    float _planetGroundRadianceEmittion;
+    float _rayleighHeightScale;
+    float _ozoneHeightScale;
+    float _mieHeightScale;
+    float _miePhaseConstant;
+    float _sunRadianceIntensity;
+    
+    glm::vec3 _mieExtinctionCoeff;
+    glm::vec3 _rayleighScatteringCoeff;
+    glm::vec3 _ozoneExtinctionCoeff;
+    glm::vec3 _mieScatteringCoeff;
+
+    // Atmosphere Debug
+    bool _saveCalculationsToTexture;
+    float _preCalculatedTexturesScale;
+
+    std::unique_ptr<AtmosphereDeferredcaster> _deferredcaster;
+
+    bool _shadowEnabled;
+    bool _hardShadows;
+    double _time;
```

Comment:
Is this value needed?  It is not used in the cpp file

Reply:
Removed.